stdlib/public/core/Bitset.swift

```diff
@@ -337,3 +337,250 @@ extension _UnsafeBitset.Word: Sequence, IteratorProtocol {
+  @inline(__always)
+  internal func uncheckedContains(_ element: Int) -> Bool {
+    _internalInvariant(isValid(element))
+    if element < Word.capacity {
+      return word0.uncheckedContains(element)
+    }
+    defer { _fixLifetime(storage) }
+    return storage!.bitset.uncheckedContains(element &- Word.capacity)
+  }
+
+  @inlinable
+  @inline(__always)
+  @discardableResult
+  internal mutating func uncheckedInsert(_ element: Int) -> Bool {
+    _internalInvariant(isValid(element))
+    let inserted: Bool
+    if element < Word.capacity {
+      inserted = word0.uncheckedInsert(element)
+    } else {
+      ensureUnique()
+      defer { _fixLifetime(storage) }
+      inserted = storage!.bitset.uncheckedInsert(element &- Word.capacity)
+    }
+    if inserted {
+      count += 1
+      _internalInvariant(count <= capacity)
+    }
+    return inserted
+  }
+
+  @inlinable
+  @inline(__always)
+  @discardableResult
+  internal mutating func uncheckedRemove(_ element: Int) -> Bool {
+    _internalInvariant(isValid(element))
+    let removed: Bool
+    if element < Word.capacity {
+      removed = word0.uncheckedRemove(element)
+    } else {
+      ensureUnique()
+      defer { _fixLifetime(storage) }
+      removed = storage!.bitset.uncheckedRemove(element &- Word.capacity)
+    }
+    if removed {
+      count -= 1
+      _internalInvariant(count >= 0)
+    }
+    return removed
+  }
+}
+
+extension _Bitset: Sequence {
+  @usableFromInline
+  internal typealias Element = Int
+
+  @inlinable
+  internal var underestimatedCount: Int {
+    @inline(__always) get { return count }
+  }
+
+  @inlinable
+  func makeIterator() -> Iterator {
+    return Iterator(self)
+  }
+
+  @usableFromInline
+  @_fixed_layout
+  internal struct Iterator: IteratorProtocol {
+    @usableFromInline
+    internal var word: Word
+    @usableFromInline
+    internal var wordIndex: Int
+    @usableFromInline
+    internal let storage: Storage?
+
+    @inlinable
+    internal init(_ bitset: _Bitset) {
+      self.word = bitset.word0
+      self.wordIndex = 0
+      self.storage = bitset.storage
+    }
+
+    @inlinable
+    internal mutating func next() -> Int? {
+      if let v = word.next() {
+        return wordIndex * Word.capacity + v
+      }
+      guard let storage = self.storage else { return nil }
+      while wordIndex < storage._wordCount {
+        word = storage._words[wordIndex]
+        // Note that _wordIndex is offset by 1 due to word0;
+        // this is why the index needs to be incremented at exactly this point.
+        wordIndex += 1
+        if let v = word.next() {
+          return wordIndex * Word.capacity + v
+        }
+      }
+      return nil
+    }
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+extension _Bitset {
+  /// A simple bitmap storage class with room for a specific number of
+  /// tail-allocated bits.
+  @_fixed_layout
+  @usableFromInline
+  internal final class Storage {
```

Comment:
Why isn't this a ManagedBuffer?

Reply:
Hm, maybe it should be! I tend to avoid it when I can help it: I find its API unappealing.

Reply:
That's a comment that probably deserves some consideration. :-) But any bare manipulation of tail elements scares me, so…

Reply:
The really scary part is all the unsafe access, and ManagedBuffer doesn't help with those. On the other hand, it adds extra layering and pollutes the class interface, which isn't always helpful.

I did get burnt by rolling my own header initialization before, which isn't an issue with ManagedBuffer. I'll have a look, and if it's not too bad, I'll switch to it!